src/python/pants/engine/native.py

```diff
@@ -419,36 +418,37 @@ def extern_create_exception(self, context_handle, msg_ptr, msg_len):
   def extern_generator_send(self, context_handle, func, arg):
     """Given a generator, send it the given value and return a response."""
     c = self._ffi.from_handle(context_handle)
+    response = self._ffi.new('PyGeneratorResponse*')
     try:
       res = c.from_value(func[0]).send(c.from_value(arg[0]))
       if isinstance(res, Get):
         # Get.
-        values = [res.subject]
-        products = [res.product]
-        tag = 2
+        response.tag = self._lib.Get
+        response.get = (
+            TypeId(c.to_id(res.product)),
+            c.to_value(res.subject),
+            c.identify(res.subject),
+          )
```

Comment:
This is a fantastic change. Is this relying on mostly the rust C representation of enums (with `tag` and (in this case) `get` fields), or is there some cffi magic going on to allow addressing unions in this deeply ergonomic way? Is there a link we could add documenting why this works to access our rust enums?

Reply:
In this case, I actually think we **don't** want to do this, as we probably want to colocate parsing the python response and producing the FFI object that represents it (so the below code is needless complexity for now). But I wrote this comment out anyway so I can link to it in #7371:

Could this part be moved into a helper method? Something like:
```python
    @memoized_property
    def _generator_response_tags_enum(self):
       class _Tags(enum([self._lib.Get, self._lib.GetMulti, self._lib.Broke, self._lib.Throw])): pass
       return _Tags

    def _set_generator_response(self, response, c, input_res, tag):
      attr, value = self._generator_response_tags_enum(tag).resolve_for_enum_variant({
        self._lib.Get: lambda: ('get', (
            TypeId(c.to_id(input_res.product)),
            c.to_value(input_res.subject),
            c.identify(input_res.subject),
          ),
        # ...
        response.tag = tag
        setattr(response, attr, value)
      })

    def extern_generator_send(...):
      # ...
              if isinstance(res, Get):
                self._set_generator_response(response, c, res, self._lib.Get)
              # ...
      return response[0]
```

Reply:
> This is a fantastic change. Is this relying on mostly the rust C representation of enums (with tag and (in this case) get fields), or is there some cffi magic going on to allow addressing unions in this deeply ergonomic way? Is there a link we could add documenting why this works to access our rust enums?

This is mostly relying on the C API that is generated by cbindgen. Kudos on integrating that.